### ISSUE TITLE
[glocaltokens.client] [!] Could not get access token

### ISSUE THREAD

**Issue by wsw70** [leikoilja/ha-google-home#680]
## Describe the bug

Upon startup of HA, I get in the console the log entry

```
domotique-hass-1  | 2023-03-11 10:16:12.036 ERROR (SyncWorker_7) [glocaltokens.client] [!] Could not get access token.
```

I do not see any adverse effect on the Google Assistant activity.

**Notes**

- I used to have GA integrated into HA a few years ago, removed it, and put it back yesterday. It works fine (my vocal requests to a google home mini are processed correctly). This log message has been there since (I belive) the first attempt of integration a few years back. It may be a leftover but I do not see where it could still be located.
- The solution mentioned at https://github.com/kiwiz/gkeepapi/issues/70#issuecomment-1213101805 does not work (I found it because logs refer to `DeviceManagementRequiredOrSyncDisabled`)

## Version of the integration

If you are unsure about the version check the [github release here](https://github.com/leikoilja/ha-google-home/releases).

```text
1.9.17
```

## Logs

```text
domotique-hass-1  | 2023-03-11 10:40:24.126 INFO (MainThread) [custom_components.google_home] 
domotique-hass-1  | -------------------------------------------------------------------
domotique-hass-1  | Google Home community driven integration
domotique-hass-1  | This is a custom integration!
domotique-hass-1  | If you have any issues with this you need to open an issue here:
domotique-hass-1  | https://github.com/leikoilja/ha-google-home/issues
domotique-hass-1  | -------------------------------------------------------------------
domotique-hass-1  | 
domotique-hass-1  | 2023-03-11 10:40:24.126 DEBUG (MainThread) [custom_components.google_home] Coordinator update interval is: 0:03:00
domotique-hass-1  | 2023-03-11 10:40:24.127 DEBUG (MainThread) [glocaltokens.client] Initializing new GLocalAuthenticationTokens instance.
domotique-hass-1  | 2023-03-11 10:40:24.127 DEBUG (MainThread) [glocaltokens.client] Set GLocalAuthenticationTokens client access_token, homegraph, access_token_date and homegraph_date to None
domotique-hass-1  | 2023-03-11 10:40:24.127 DEBUG (MainThread) [glocaltokens.client] Set GLocalAuthenticationTokens client username to "w**********", password to "t***************", master_token to "a******************************************************************************************************************************************************************************************************************************" and android_id to b***************
domotique-hass-1  | 2023-03-11 10:40:24.128 DEBUG (SyncWorker_8) [glocaltokens.client] Initializing models list if empty...
domotique-hass-1  | 2023-03-11 10:40:24.129 DEBUG (SyncWorker_8) [glocaltokens.client] Forcing homegraph reload
domotique-hass-1  | 2023-03-11 10:40:24.129 DEBUG (SyncWorker_8) [glocaltokens.client] Invalidated homegraph
domotique-hass-1  | 2023-03-11 10:40:24.129 DEBUG (SyncWorker_8) [glocaltokens.client] Getting homegraph...
domotique-hass-1  | 2023-03-11 10:40:24.129 DEBUG (SyncWorker_8) [glocaltokens.client] There is no stored homegraph, or it has expired, getting a new one...
domotique-hass-1  | 2023-03-11 10:40:24.129 DEBUG (SyncWorker_8) [glocaltokens.client] There is no access_token stored, or it has expired, getting a new one...
domotique-hass-1  | 2023-03-11 10:40:24.129 DEBUG (SyncWorker_8) [glocaltokens.client] Master token: a******************************************************************************************************************************************************************************************************************************
domotique-hass-1  | 2023-03-11 10:40:24.307 ERROR (SyncWorker_8) [glocaltokens.client] [!] Could not get access token.
domotique-hass-1  | 2023-03-11 10:40:24.310 DEBUG (SyncWorker_8) [glocaltokens.client] Request response: {'Error': 'DeviceManagementRequiredOrSyncDisabled'}
domotique-hass-1  | 2023-03-11 10:40:24.310 DEBUG (SyncWorker_8) [glocaltokens.client] [GRPC] Unable to obtain access token.
domotique-hass-1  | 2023-03-11 10:40:24.310 DEBUG (SyncWorker_8) [glocaltokens.client] Failed to fetch homegraph
domotique-hass-1  | 2023-03-11 10:40:24.392 DEBUG (MainThread) [custom_components.google_home] Finished fetching sensor data in 0.265 seconds (success: True)
```

**Comment by leikoilja** [comment]
@wsw70, please see WIP here https://github.com/leikoilja/ha-google-home/issues/599#issuecomment-1458000920
 Closing this ticket since it's a duplicate

**Comment by wsw70** [comment]
@leikoilja I am not sure this is the same issue - my integration works fine.

Additionally I think that this is a leftover of a previous (long uninstalled) version.

**Comment by leikoilja** [comment]
Sorry, @wsw70, you're right, i overlooked that you are hit back with `DeviceManagementRequiredOrSyncDisabled` from google APIs. Tbh, have not yet seen this type of issue before.
From your logs the code is trying to get the access token using the master token. But as we see during that - the google APIs is returning this mysterious `DeviceManagementRequiredOrSyncDisabled`.
The issue has nothing to do with `ha-google-home` package and unless we can replicate it, I am afraid you are left on your own.

What you have seen https://github.com/kiwiz/gkeepapi/issues/70#issuecomment-1213101805 is probably the only reference on that issue. Also see https://gkeepapi.readthedocs.io/en/latest/#toc-entry-34 as it gives some more tips to try debugging and resolving it. As they have mentioned 

> This is due to the enforcement of Android device policies on your G-Suite account.
sounds quite descriptive towards the error message you are seeing

**Comment by wsw70** [comment]
Yes, I see that the issue is not widespread. The bizarre thing is that the integration works fine. This really seems to be some leftover from the previous (a few years old) attempt and uninstallation.

**Comment by leikoilja** [comment]
@wsw70 when you say that integration works fine - is it able to pull the data from your google home devices? i wonder how, if the logs was showing that it failed to authenticate 🤔

**Comment by wsw70** [comment]
> @wsw70 when you say that integration works fine - is it able to pull the data from your google home devices? i wonder how, if the logs was showing that it failed to authenticate 🤔

- when I say to my home mini "switch of the lights in livingroom center" (where "livingroom center" is a specific area) it works
- when I say "switch on square" (where "square" is the name of a lamp) it works

→ so the communication from GH to HA is OK (I took examples of specific devices or areas to make sure that I indeed target something I manually named, I do not know GH enough yet so I am not sure whether it would "extrapolate" any orders to target the best match)

- the state of the devices as seen in the Android Home application are correct (yellow when a light is actually on, and gry otherwise)
- when I modified the domains to expose in `configuration.yml`, they were immediately synchronized with the Home app (I removed the domain `switch` and everything that was configured as a switch got removed from Home, with relevant logs in the Home app)

→ so the communication from HA to GH works

I was quite surprised as well when I saw the logs. Since the `glocatokens` error **predates** the **last** installation of your integration and that I belive(d) that this is a leftover, I was expecting two authentications: a successful one from the current integration, and a failed one from the leftovers.

**Comment by leikoilja** [comment]
@wsw70 i am afraid you are mixing some things up a little. 

From your description, the integration that works for you is the integration that syncs the state of home assistant to your google assistant. Most likely it's https://www.home-assistant.io/integrations/google_assistant/ 

Our integration `ha-google-home` takes care of local device communication with your google home speakers to load and control their alarms, timers, do not disturb mode, volume etc. 

The error you see in logs prevents `ha-google-home` from working and pulling the data from the devices. Although the error is not caused by the `ha-google-home` integration, most likely it's due to some google account configurations (as mentioned above)

**Comment by wsw70** [comment]
AHHHHH... thanks a lot for making it clear to me. I was not seeing the difference.

So yes, I have the google assistant configuration configured and it works.

I then realized that I also had `ha-google-home` configured - badly as it turns out because of the error. I removed it, it was auto-discovered, I configured it (this time correctly) and it shows my Google Home devices. And a bunch of debug logs where everything is fine.

I will have to understand what exact use I can make out of this - especially since I, for now, only use my mini homes as "order takers" (switch this off, move the rollers up) but never anything fancier. Your integration will force m to get deeper (thank you :))

Again, thanks for having taken the time to explain my misunderstanding!
